recipes/autoconf/all/test_package/conanfile.py

```diff
@@ -1,47 +1,53 @@
+    generators = "AutotoolsDeps", "AutotoolsToolchain", "VirtualBuildEnv"
     test_type = "explicit"
 
     @property
     def _settings_build(self):
+        # TODO: Remove for Conan v2
         return getattr(self, "settings_build", self.settings)
 
+    @property
+    def win_bash(self):
+        return self._settings_build.os == "Windows"
+
     def build_requirements(self):
-        self.build_requires(self.tested_reference_str)
-        if self._settings_build.os == "Windows" and not tools.get_env("CONAN_BASH_PATH"):
-            self.build_requires("msys2/cci.latest")
-
-    @contextlib.contextmanager
-    def _build_context(self):
-        if is_msvc(self):
-            with tools.vcvars(self):
-                with tools.environment_append({"CC": "cl -nologo", "CXX": "cl -nologo",}):
-                    yield
-        else:
-            yield
+        if self._settings_build.os == "Windows" and not self.conf.get("tools.microsoft.bash:path", default=False, check_type=bool):
+            self.tool_requires("msys2/cci.latest")  # The conf `tools.microsoft.bash:path` and `tools.microsoft.bash:subsystem` aren't injected for test_package
+        self.tool_requires(self.tested_reference_str)
 
     def build(self):
+        if self._settings_build.os == "Windows" and not self.conf.get("tools.microsoft.bash:path", default=False, check_type=bool):
+            return  # autoconf needs a bash if there isn't a bash no need to build
+
         for src in self.exports_sources:
-            shutil.copy(os.path.join(self.source_folder, src), self.build_folder)
-        self.run("{} --verbose".format(os.environ["AUTOCONF"]),
-                 win_bash=tools.os_info.is_windows, run_environment=True)
-        self.run("{} --help".format(os.path.join(self.build_folder, "configure").replace("\\", "/")),
-                 win_bash=tools.os_info.is_windows, run_environment=True)
-        autotools = AutoToolsBuildEnvironment(self, win_bash=tools.os_info.is_windows)
-        with self._build_context():
-            autotools.configure()
-            autotools.make()
+            shutil.copy(path.join(self.source_folder, src), self.build_folder)
+
+        self.run(f"autoconf --verbose", env="conanbuild")
+
+        autotools = Autotools(self)
+        autotools.configure(build_script_folder=self.build_folder)
+        autotools.make()
 
     def test(self):
-        if not tools.cross_building(self):
-            self.run(os.path.join(".", "test_package"), run_environment=True)
+        if self._settings_build.os == "Windows" and not self.conf.get("tools.microsoft.bash:path", default=False, check_type=bool):
+            return  # autoconf needs a bash if there isn't a bash no need to build
+
+        if can_run(self):
+            ext = ".exe" if self.settings.os == "Windows" else ""
+            test_cmd = unix_path(self, path.join(self.build_folder, f"test_package{ext}"))
+
+            self.run(test_cmd, env="conanbuild")
```

Comment:
not need to protect with can_run, since it's executed in build scope.

Reply:
Should this be run scope? Seems odd to be using build scope in a "can the test run" scenario 🤔

Reply:
yes it should be run scope